modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/operations/util/SyncmlCredentials.java

```diff
@@ -62,9 +61,9 @@ public String generateRST(String username, String password) throws UnsupportedEn
             String usrPwdNonce = usrPwdHash + ":" + nonce;
             usrPwdNonceHash = Base64.encodeBase64String(digest.digest(usrPwdNonce.getBytes(Constants.UTF_8)));
         } catch (UnsupportedEncodingException e) {
-            throw new UnsupportedEncodingException("Problem occurred in encoding credentials data.");
+            throw new SyncmlMessageFormatException("Problem occurred in encoding credentials data.", e);
         } catch (NoSuchAlgorithmException e) {
-            throw new NoSuchAlgorithmException("Problem occurred in generating password hash.", e);
+            throw new SyncmlMessageFormatException("Problem occurred in generating password hash.", e);
```

Comment:
not the appropriate message to be passed as this exception occurs not by generating the password hash, but by MessageDigest.getInstance(Constants.MD5). So the message should be like "Application environment does not have an implementation available/configured for the requested algorithm".